test/integration/protocol_integration_test.cc

```diff
@@ -4921,4 +4921,64 @@ TEST_P(DownstreamProtocolIntegrationTest, UnknownPseudoHeader) {
   }
 }
 
+// This test performs DNS refreshing very fast also keep changing DNS results.
+// It is trying to reproduce a R/W race for logical_dns_cluster.
```

Comment:
please write an Issue for this issue, and reference it here. You might even say something like:
```
// Reproduces a race from #xxxx that is fixed in #32716, by mocking the DNS resolver
// to rapidly change dns resolutions for an ip address.
```

Reply:
Please also add instructions on how to reproduce the test, i.e. TSAN build and repeated runs

Reply:
The test command to reproduce the issue:

bazel test  test/integration:protocol_integration_test  --config=clang-tsan  --config=remote

This happens consistently in my VM.  If the race somehow is not reproducible in an extremely fast VM, then one can add some delay , like a sleep(1),  here to force the race: https://github.com/envoyproxy/envoy/blob/d82b5a02befbff3b2119b309e865f7b44d207907/source/common/network/happy_eyeballs_connection_impl.cc#L81.